### PR TITLE
Palladio 495

### PR DESCRIPTION
**Pull request by rapp-fzi**
Please review and merge by considering the review notes in the ticket https://jira.palladio-simulator.com/browse/PALLADIO-495

### CODE REVIEW

File: bundles/org.palladiosimulator.simulizar.action/src-man/org/palladiosimulator/simulizar/action/interpreter/ActionRuntimeState.java [hunk 1]
```diff
@@ -69,7 +73,7 @@ public void setRuntimeStateModel(AbstractSimuLizarRuntimeState passedState) {
         private ControllerCallInputVariableUsageCollection controllerCallVariableUsages = EMPTY_VARIABLE_USAGE_COLLECTION;
         private boolean isAsync = false;
 
-        private TransientEffectInterpreterBuilder(RoleSet roleSet, AdaptationBehaviorRepository repository) {
+        private TransientEffectInterpreterBuilder(RoleSet roleSet, AdaptationBehaviorRepository repository                ) {
```
Comment (sdkrach) [inline bundles/org.palladiosimulator.simulizar.action/src-man/org/palladiosimulator/simulizar/action/interpreter/ActionRuntimeState.java:76]: Empty space

File: bundles/org.palladiosimulator.simulizar.action/model/actions.ecore [hunk 2]
```diff
@@ -18,66 +18,74 @@
...
-          <details key="body" value="org.palladiosimulator.simulizar.action.interpreter.ActionRuntimeState.getInterpreterBuilder(affectedRoleSet, getRepository()).isAsync(asyncExecutionContext).build().doSwitch(this);&#xD;&#xA;return asyncExecutionContext;&#xD;&#xA;"/>
+          <details key="body" value="org.palladiosimulator.simulizar.action.interpreter.ActionRuntimeState.getInterpreterBuilder(affectedRoleSet, getRepository()).isAsync(asyncExecutionContext).build(resourceTableManager).doSwitch(this);&#xA;return asyncExecutionContext;&#xA;"/>
           <details key="documentation" value="Triggers the asynchronous execution of this AdaptationBehavior for the given RoleSet in the given ExecutionContext.&#xD;&#xA;To conform to ome overloaded versions of this method, this context  is also returned.&#xD;&#xA;It should be passed to synchronous adaptation behavior executions that are done within."/>
         </eAnnotations>
         <eParameters name="affectedRoleSet" eType="#//instance/RoleSet"/>
         <eParameters name="asyncExecutionContext" eType="#//context/ExecutionContext"/>
+        <eParameters name="resourceTableManager" eType="#//core/IResourceTableManager"/>
```
Comment (sdkrach) [inline bundles/org.palladiosimulator.simulizar.action/model/actions.ecore:88]: This is really problematic, as all action specifications in qvto will break. I would strongly suggest resolving the manager somewhere in the execution context. Furthermore, the parameter does not make sense to be carried into the QVTo world.

**Review by sdkrach — CHANGES_REQUESTED** [review]
In principle the change looks fine. There ist however a huge problem with introducing an additional parameter to the executeAsync of the ExecutionContext. This will break the entire Actions Library.

File: bundles/org.palladiosimulator.simulizar.action/src-man/org/palladiosimulator/simulizar/action/interpreter/ActionRuntimeState.java (hunk 1 above)
Comment (rapp-fzi) [reply bundles/org.palladiosimulator.simulizar.action/src-man/org/palladiosimulator/simulizar/action/interpreter/ActionRuntimeState.java:76]: worked in review finding; reformatted file using Palladio code formatter

File: bundles/org.palladiosimulator.simulizar.action/model/actions.ecore (hunk 2 above)
Comment (rapp-fzi) [reply bundles/org.palladiosimulator.simulizar.action/model/actions.ecore:88]: The current solution does not break the qvto scripts; by passing the resourceTableManager as 'configuration property resourceTableManager : IResourceTableManager;' to the respective qvto scripts (only script ' Palladio-Analyzer-SimuLizar\bundles\ExampleModels\TransientEffects\org.palladiosimulator.simulizar.examples.mediastore\reconfigurations_active\horizontalscaling.qvto' is affected by this change), the scripts are still executable; for sure, passing the parameter to the script is just a temporarily workaround to guarantee, that the scripts are still executable. Thus I created the follow-up ticket https://jira.palladio-simulator.com/browse/SIMULIZAR-130 to address issue.

**Review by sdkrach — APPROVED** [review]
In view of the follow up issue the change so far LGTM.

**Pull request merged**